erpnext/stock/doctype/warehouse/warehouse.py

```diff
@@ -199,4 +229,54 @@ def get_warehouses_based_on_account(account, company=None):
 		frappe.throw(_("Warehouse not found against the account {0}")
 			.format(account))
 
-	return warehouses
\ No newline at end of file
+	return warehouses
+
+def get_warehouse_list(doctype, txt, filters, limit_start, limit_page_length=20, order_by="modified"):
+	"""Getting list of warehouse and there items details from Warehouse and Bin doctype."""
+	user = frappe.session.user
+	customers = []
+	docs = []
+	# Checking user have 'Warehouse' permission or not.
+	if frappe.has_permission("Warehouse"):
```

Comment:
Lets remove this, just saw, `allow_guest_to_view` in doctype json

Reply:
sorry allow_guest to view would be 0.
as only user with customer role allow to see their warehouse.